src/gtfs/read.rs

```diff
@@ -441,7 +441,27 @@ where
                 (false, true) => Some(StopTimePrecision::Approximate),
                 (true, true) => Some(StopTimePrecision::Estimated),
             };
-
+            if stop_time.pickup_type == 2 || stop_time.drop_off_type == 2 {
+                if let Some(comment_id) = manage_odt_comment_from_stop_time(
+                    &mut collections.comments,
+                    &mut collections.companies,
+                    on_demand_transport_comment.clone(),
+                    &vj.company_id,
+                ) {
+                    let stop_time_id = format!(
+                        "{}-{}",
+                        &stop_time.trip_id,
+                        &stop_time.stop_sequence.to_string()
+                    );
+                    let comment_idx = collections.comments.get_idx(&comment_id).unwrap();
+                    collections
+                        .stop_time_ids
+                        .insert((vj_idx, stop_time.stop_sequence), stop_time_id);
+                    collections
+                        .stop_time_comments
+                        .insert((vj_idx, stop_time.stop_sequence), comment_idx);
+                }
```

Comment:
I believe this whole block should be part of the `manage_odt_comment_from_stop_time` function for multiple reasons:
- first of all, this `manage_stop_times` function is already pretty big
- the current `manage_odt_comment_from_stop_time` function is creating a `Comment` where you could get back directly the `Idx<>`. But instead, the function push the `Comment` on the `CollectionWithId` and then only after, you search again for it with a `get_idx`.

Reply:
For this purpose, i need to 2 more parameters for manage_odt_comment_from_stop_time (collections.stop_time_ids and collections.stop_time_comment) and lots of parameters are considered bad style and reduce readability (Clippy Lints)

Reply:
If you need multiple fields out of `collections`, then why not sending the an entire mutable reference on `collections`?

Reply:
collections.vehicle_journeys is borrowing before manage_odt_comment_from_stop_time (collections.vehicle_journeys.index_mut(vj_idx).